src/text-field-container.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const React = require('react');
+const mdl = require('material-design-lite/material.min');
+const classnames = require('classnames');
+
+const baseClasses = {
+  'mdl-textfield': true,
+  'mdl-js-textfield': true
+};
+
+class TextFieldContainer extends React.Component {
+
+  getChildContext(){
+  return {
+      id: '_' + Math.random().toString(36).slice(2)
+    };
+  }
+
+  componentDidMount(){
+    const node = this._element;
+    mdl.upgradeElement(node, 'MaterialTextfield');
+  }
+
+  componentWillUnmount(){
+    const node = this._element;
+    mdl.downgradeElements(node);
+  }
+
+  render(){
+    const {
+      children,
+      expandable,
+      icon,
+      floating,
+      className
+    } = this.props;
+
+    const classes = classnames(baseClasses, {
+      'mdl-textfield--expandable': expandable,
+      'mdl-button mdl-js-button mdl-button--icon': icon,
```

Comment:
By applying it to the parent element it updates the entire element.  I ran tests to make sure on this one.  The code for upgradeElement describes this bettter.